tmt/hardware.py

```diff
@@ -0,0 +1,1176 @@
+
+    if 'network' in spec:
+        group.constraints += [_parse_networks(spec['network'])]
+
+    if 'hostname' in spec:
+        group.constraints += [
+            Constraint.from_specification(
+                'hostname',
+                spec['hostname'],
+                as_quantity=False)]
+
+    if 'tpm' in spec:
+        group.constraints += [_parse_tpm(spec['tpm'])]
+
+    if 'virtualization' in spec:
+        group.constraints += [_parse_virtualization(spec['virtualization'])]
+
+    return group
+
+
+@ungroupify
+def _parse_and(spec: Spec) -> BaseConstraint:
+    """
+    Parse an ``and`` clause holding one or more subblocks or constraints.
+
+    :param spec: raw constraint block specification.
+    :returns: block representation as :py:class:`BaseConstraint` or one of its subclasses.
+    """
+
+    group = And()
+
+    group.constraints += [
+        _parse_block(member)
+        for member in spec
+        ]
+
+    return group
+
+
+@ungroupify
+def _parse_or(spec: Spec) -> BaseConstraint:
+    """
+    Parse an ``or`` clause holding one or more subblocks or constraints.
+
+    :param spec: raw constraint block specification.
+    :returns: block representation as :py:class:`BaseConstraint` or one of its subclasses.
+    """
+
+    group = Or()
+
+    group.constraints += [
+        _parse_block(member)
+        for member in spec
+        ]
+
+    return group
+
+
+@ungroupify
+def _parse_block(spec: Spec) -> BaseConstraint:
+    """
+    Parse a generic block of HW constraints - may contain ``and`` and ``or``
+    subblocks and actual constraints.
+
+    :param spec: raw constraint block specification.
+    :returns: block representation as :py:class:`BaseConstraint` or one of its
+    subclasses.
+    """
+
+    if 'and' in spec:
+        return _parse_and(spec['and'])
+
+    if 'or' in spec:
+        return _parse_or(spec['or'])
+
+    return _parse_generic_spec(spec)
+
+
+def parse_hw_requirements(spec: Spec) -> BaseConstraint:
+    """
+    Convert raw specification of HW constraints to our internal representation.
+
+    :param spec: raw constraints specification as stored in an environment.
+    :returns: root of HW constraints tree.
+    """
+
+    return _parse_block(spec)
+
+
+@dataclasses.dataclass
+class Hardware(tmt.utils.SpecBasedContainer):
+    constraint: Optional[BaseConstraint]
+    spec: Spec
+
+    @classmethod
+    def from_spec(cls: Type['Hardware'], spec: Spec) -> 'Hardware':
+        if not spec:
+            return Hardware(constraint=None, spec=spec)
+
+        return Hardware(
+            constraint=parse_hw_requirements(spec),
+            spec=spec
+            )
+
+    def to_spec(self) -> Dict[str, Any]:
+        return self.spec  # type: ignore[no-any-return]
+
+    def to_minimal_spec(self) -> Dict[str, Any]:
+        return self.spec  # type: ignore[no-any-return]
+
+    def show_support(self, check: Callable[[Constraint], bool], logger: tmt.log.Logger) -> None:
```

Comment:
```suggestion
    def check_support(self, check: Callable[[Constraint], bool], logger: tmt.log.Logger) -> None:
```

Wouldn't `check` be a bit more appropriate as the main goal of the method seems to be verifying the constraints?

Reply:
Ok, understood. What about `examine_support()` or `explore_support()`?

Reply:
Renamed to `report_support()` in dc5b7deb394fcbec89dd5bc4197ad4c566022076